test/pytest/mockwebserver.py

```diff
@@ -0,0 +1,492 @@
+                capabilities=['multi', 'credentials', 'binary'],
+                system={'version': '0'}
+            )
+
+            # receive "init" from the websocket
+            try:
+                assert await self.receive_json() == {'command': 'init', 'version': 1}
+            except (TypeError, json.JSONDecodeError, AssertionError) as exc:
+                raise CockpitProtocolError('expected init message') from exc
+
+            # send "init" to the bridge
+            # TODO: explicit-superuser handling
+            transport.data_received(b'\n' + json.dumps({
+                "command": "init",
+                "version": 1,
+                "host": "localhost"
+            }).encode())
+
+            write_task = asyncio.create_task(self.process_outgoing_queue(self.outgoing_queue))
+
+            try:
+                async for msg in self:
+                    if msg.type == aiohttp.WSMsgType.TEXT:
+                        frame = msg.data
+                        if frame.startswith('\n'):
+                            control = json.loads(frame)
+                            command = get_str(control, 'command')
+                            channel = get_str(control, 'channel', None)
+                            if command == 'open':
+                                if channel is None:
+                                    raise CockpitProtocolError('open message without channel')
+                                binary = get_enum(control, 'binary', ['raw'], None) == 'raw'
+                                transport.register_origin(binary_origin if binary else text_origin, channel)
+                        transport.data_received(frame.encode())
+                    elif msg.type == aiohttp.WSMsgType.BINARY:
+                        transport.data_received(msg.data)
+                    else:
+                        raise CockpitProtocolError(f'strange websocket message {msg!s}')
+            finally:
+                self.outgoing_queue.put_nowait(None)
+                await write_task
+
+        except CockpitProblem as exc:
+            if not self.closed:
+                await self.send_control(exc.get_attrs(), command='close')
+
+
+routes = web.RouteTableDef()
+
+
+@routes.get(r'/favicon.ico')
+async def favicon_ico(request: web.Request) -> web.FileResponse:
+    del request
+    return web.FileResponse('src/branding/default/favicon.ico')
+
+
+SPLIT_UTF8_FRAMES = [
+    b"initial",
+    # split an é in the middle
+    b"first half \xc3",
+    b"\xa9 second half",
+    b"final"
+]
+
+
+@routes.get(r'/mock/expect-warnings')
+@routes.get(r'/mock/dont-expect-warnings')
+async def mock_expect_warnings(_request: web.Request) -> web.Response:
+    # no op — only for compatibility with C test-server
+    return web.Response(status=200, text='OK')
+
+
+@routes.get(r'/mock/info')
+async def mock_info(_request: web.Request) -> web.Response:
+    return web.json_response({
+        'pybridge': True,
```

Comment:
Dumb unrelated question, why do we still have this? We only have a pybridge

Reply:
Right, I sent #21583 to clean it up (that's the level of engineering I'm capable of today..)